cmake/Modules/FindTPLARMPL.cmake

```diff
@@ -0,0 +1,32 @@
+IF (ARMPL_LIBRARY_DIRS AND ARMPL_LIBRARIES)
```

Comment:
These changes look reasonable, could you post any configure and test output from a system where you tested this? (I'm guessing maybe this was on mayer)? If this is being developed on mayer we can set up a nightly test with these tpls once this goes in

Reply:
Thanks for your review, @ndellingwood !

## Expected configure fail
```bash
$ ../cm_generate_makefile.bash --with-tpls=armpl --with-scalars=float --arch=A64FX --with-serial --with-openmp --kokkos-path=$HOME/KOKKOS.base/kokkos --kokkos-prefix=$HOME/KOKKOS.base/kokkos/install -kokkoskernels-path=$HOME/KOKKOS.base/kokkos-kernels --prefix=$HOME/KOKKOS.base/kokkos-kernels/install
<snip>
CMake Error at cmake/Modules/FindTPLARMPL.cmake:21 (FIND_PACKAGE):
  By not providing "FindARMPL.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "ARMPL", but
  CMake did not find one.

  Could not find a package configuration file provided by "ARMPL" with any of
  the following names:

    ARMPLConfig.cmake
    armpl-config.cmake

  Add the installation prefix of "ARMPL" to CMAKE_PREFIX_PATH or set
  "ARMPL_DIR" to a directory containing one of the above files.  If "ARMPL"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake/kokkoskernels_tpls.cmake:363 (FIND_PACKAGE)
  cmake/kokkoskernels_tpls.cmake:485 (KOKKOSKERNELS_IMPORT_TPL)
  CMakeLists.txt:175 (INCLUDE)


-- Configuring incomplete, errors occurred!
```

## Expected configure pass
```bash
$ rm -rf CMake*
$ module load armpl/20.3.0
$ ../cm_generate_makefile.bash --with-tpls=armpl --with-scalars=float --arch=A64FX --with-serial --with-openmp --kokkos-path=$HOME/KOKKOS.base/kokkos --kokkos-prefix=$HOME/KOKKOS.base/kokkos/install -kokkoskernels-path=$HOME/KOKKOS.base/kokkos-kernels --prefix=$HOME/KOKKOS.base/kokkos-kernels/install
<snip>
-- Found Kokkos at /path/to/kokkos/install/lib64/cmake/Kokkos
-- Found TPLARMPL: /path/to/include

=======================
KokkosKernels ETI Types
   Devices:  <OpenMP,HostSpace>;<Serial,HostSpace>
   Scalars:  double;float
   Ordinals: int
   Offsets:  int;size_t
   Layouts:  LayoutLeft

KokkosKernels TPLs
   ARMPL:       /path/to/libamath.so;/path/to/libarmpl.so
=======================
```

## Test suite results
```bash
$ make -j8
$ # TODO
```

Let's discuss setting up nightly tests with this tpl at the next meeting.